### PR TITLE
Add Spark's date function to Parser.FUNCTIONS

### PR DESCRIPTION
**Pull request by vegarsti**
Similar to https://github.com/tobymao/sqlglot/pull/1454, and encouraged by #1455.

### CODE REVIEW

**Review by GeorgeSittas — APPROVED** [review]
Thanks for the PR! You can fix the style using `make style`, but as a rule of thumb just do `make check` before committing.

LGTM -- Are these `timestamp`, `date` functions only available in Spark SQL, or in Hive as well? We usually add functionality to Hive because Spark inherits it, but I'm not sure if we should in this case.

**Comment by vegarsti** [comment]
Ah, thanks, will do that going forward!

Great question  -- will check!

**Comment by vegarsti** [comment]
Doesn't look like these are in Hive, no: https://cwiki.apache.org/confluence/plugins/servlet/mobile?contentId=27362046#LanguageManualUDF-DateFunctions

I don't have a hive db i can test on, though, so don't know for sure

**Comment by GeorgeSittas** [comment]
Ok no worries, we can be conservative and leave it as is then.

**Comment by vegarsti** [comment]
Btw just so you know, not at the computer for a few hours, so  can't fix the syntax yet.

File: sqlglot/dialects/spark.py [hunk 1]
```diff
@@ -89,6 +89,9 @@ class Parser(Hive.Parser):
             "WEEKOFYEAR": lambda args: exp.WeekOfYear(
                 this=exp.TsOrDsToDate(this=seq_get(args, 0)),
             ),
+            "DATE": lambda args: exp.Cast(
+                this=seq_get(args, 0), to=exp.DataType.build("date")
+            ),
```
Comment (GeorgeSittas) [inline sqlglot/dialects/spark.py:94]:
```suggestion
            "DATE": lambda args: exp.Cast(this=seq_get(args, 0), to=exp.DataType.build("date")),
```

**Comment by GeorgeSittas** [comment]
> Btw just so you know, not at the computer for a few hours, so can't fix the style yet.

No problem, went ahead and fixed it for you.

**Comment by vegarsti** [comment]
Thanks!!

**Pull request merged**